fpgalib/tools/ghz_fpgas.lua

```diff
@@ -0,0 +1,206 @@
+-- Protocol definitions and packet dissectors for GHz FPGAs
+--
+-- To use this in wireshark, you can copy this file into your
+-- local wireshark plugin directory. You can find the plugin
+-- directory for your platform by looking at the wireshark
+-- help menu. This is typically something like:
+--
+--  windows: C:/Users/<user>/AppData/Roaming/Wireshark/plugins/
+--  linux: ~/.wireshark/plugins/
+--
+-- Note that this uses heuristic dissectors to identify the
+-- fpga packets. Heuristic dissectors were first exposed to
+-- the lua API in version 1.11.3. On some linux distributions,
+-- the packaged version of wireshark is too old, so you may
+-- have to build from source.
+--
+-- For details of the wireshark lua API, see:
+-- https://www.wireshark.org/docs/wsdg_html_chunked/lua_module_Proto.html
+
+local p_dac = Proto("ghz_dac", "GHz DAC")
+local f_dac_dest = ProtoField.ether("ghz_dac.dest", "Destination")
```

Comment:
yeah, the first string (`ghz_dac.dest`) is sort of an internal thing as far as I can tell, while the second string (`Destination`) shows up in the tree view when expanding the dissected packet tree. It also seems that the first string must be unique across all fields, and that a given field object can only be added to one protocol. This is annoying because the ethernet source and destination fields clearly ought to work for both DAC and ADC packets, but wireshark doesn't allow that AFAICT.